apps/meteor/app/apps/client/gameCenter/GameCenter.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import type { ReactElement, SyntheticEvent } from 'react';
+import type { IExternalComponent } from '@rocket.chat/apps-engine/definition/externalComponent';
+import { useMutableCallback, useSafely } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+
+import { useTabBarClose } from '../../../../client/views/room/providers/ToolboxProvider';
+import GameCenterContainer from './GameCenterContainer';
+import GameCenterList from './GameCenterList';
+
+export type IGame = IExternalComponent;
+
+const GameCenter = (): ReactElement => {
+	const [games, setGames] = useState<IGame[]>();
+	const [isOpened, setIsOpened] = useState<boolean>(false);
+	const [openedGame, setOpenedGame] = useState<IGame>();
+	const [isLoading, setLoading] = useSafely(useState(false));
+
+	const closeTabBar = useTabBarClose();
+	const prevent = (e: SyntheticEvent): void => {
+		if (e) {
+			(e.nativeEvent || e).stopImmediatePropagation();
+			e.stopPropagation();
+			e.preventDefault();
+		}
+	};
+
+	const getGamesEndpoint = useEndpoint('GET', '/apps/externalComponents');
+
+	useEffect(() => {
+		async function getGames() {
+			const { externalComponents } = await getGamesEndpoint();
+			setGames(externalComponents);
+		}
+
+		if (!games) {
+			setLoading(true);
+			getGames();
+			setLoading(false);
+		}
+	}, [games, getGamesEndpoint, setLoading]);
```

Comment:
Please replace this effect with a `useQuery` call. It enables some additional goodies for async data fetching, like loading/error/success states and caching.